fix(http): allow to combine FetchBackend and  InMemory services in the providers

JeanMeche opened the pull request:
When using `withFetch`,  the `PRIMARY_HTTP_BACKEND` token is set. 

The InMemory Backend services will also set that token. 

This means that providers order will matter and the latest on the list will be the one instantiated.

----
cc @alan-agius4, this is related to [your comment](https://github.com/angular/angular/pull/52398#discussion_r1373620361) on #52398

alan-agius4 commented on packages/misc/angular-in-memory-web-api/src/http-client-in-memory-web-api-module.ts:
```diff
@@ -8,19 +8,12 @@
 
 import {XhrFactory} from '@angular/common';
 import {HttpBackend} from '@angular/common/http';
+import {PRIMARY_HTTP_BACKEND} from '@angular/common/http/src/interceptor';
```
Deeps imports will break during runtime.

You’d need to export the symbol as private .

jessicajaniuk reviewed (approved):
LGTM!

dylhunn commented:
Caretaker -- the aio failure looks spurious to me

alxhub commented:
This PR was merged into the repository by commit 49b037f8116d56520035a1cc0797b65f1b0e3ad9.

JeanMeche committed:
fix(http): Don't override the backend when using the InMemoryWebAPI

When using `withFetch`,  the `PRIMARY_HTTP_BACKEND` token is set.

The InMemory Backend services will also set that token.

This means that providers order will matter and the latest on the list will be the one instantiated

JeanMeche commented:
The initial merge was reverted in #52425 because of test regressions. 

Those are now fixed.

jessicajaniuk commented:
This PR was merged into the repository by commit 291ba38623cfee60d0b734c3ffbb7b5f3fae06a6.
